### ISSUE TITLE
Enabled LDM manager intermittently makes network connections on plug/unplug from RC

### ISSUE THREAD

**Issue by dokazaki** [dji-sdk/Mobile-SDK-iOS#249]
The LDM Manager is much improved in the latest release, but still has some bugs in it that occur when plugging or unplugging the iPad from the RC.

Our application enables LDM at app start, before calling SDK app init and the logs show LDM is enabled. At no time do we ever get notification that LDM becomes disabled. However, we still detect network connected aside from the initial app registration in the following cases:

- Uninstall, re-install, then run the app
  - the app enables LDM and registers the app (one network connect to dev.dji.com as expected)
  - power on the RC, but do not power on the aircraft
  - connect the iPad to the RC
  - this causes one POST to api.dji-services.com
    - in one case, this also started a persistent CONNECT to api.dji-services.com that would not end until the app was killed and the RC unplugged
- Uninstall, re-install, then run the app
  - the app enables LDM and registers the app (one network connect to dev.dji.com as expected)
  - connect the iPad to the unpowered RC
  - power on the RC
  - this sometimes causes CONNECT to 2 skypixel addresses
    - CONNECT to www.skypixel.com - ends after 9ms
    - CONNECT to app-service.skypixel.com lasts for 76seconds before ending
  - otherwise there is one GET and one POST to api.dji-services.com
  - power on the aircraft - no new network connections
  - unplug the iPad from the RC
  - this sometimes causes one POST to api.dji-services.com
- Kill app, **reboot iPad**, run app
  - the app enables LDM and registers the app (one network connect to dev.dji.com as expected)
  - connect the iPad to the unpowered RC
  - power on the RC
  - this causes CONNECT to:
    - CONNECT to active.dji.com - 4 connects, lasting 9 - 15 seconds each
    - CONNECT to mydjiflight.dji.com - 2 connects, lasting ~60 seconds each
    - sometimes
       - CONNECT to www.skypixel.com
       - CONNECT to app-service.skypixel.com
  - also there is one GET and sometimes one POST to api.dji-services.com
- Kill the app, restart it while connected to RC with RC and aircraft powered on
  - the app enables LDM and registers the app (one network connect to dev.dji.com as expected)
  - unplug the iPad from the RC
  - this causes one POST to api.dji-services.com

**Execution environment:**
DJI Phantom 4 Pro - outside (no sim)
iPad Pro 10.5"
iOS 11.3, 11.3.1
DJI iOS Mobile SDK 4.5.1
Xcode 9.3

**Comment by XietongLU** [comment]
Thanks for reporting this. That will be very helpful for us to improve LDM in SDK.

**Comment by dokazaki** [comment]
@XietongLU - I updated the issue - it appears the SDK is relying on memory cache for LDM. When the iPad is rebooted, the SDK makes network connections when connected to RC and RC is powered on, even after LDM is enabled and after SDK registration.

**Comment by dokazaki** [comment]
@XietongLU - I found an older copy of our app downloaded from TestFlight on the iPad I was using for testing, which was linked to SDK 4.4. This was the root cause to the network leakage reported by the network proxy/monitor.

Even when the debug build of the app is running, iOS chooses to launch the "released" app when the RC is plugged into the lightning port.

When I deleted the older app, the network proxy did not detect any of the above described network "leaks".  So it looks like LDM in 4.5.1 is actually working as documented.
